From 487209d87680df904233b111bb98a7d05e092ebf Mon Sep 17 00:00:00 2001
From: Deokgyu Yang <secugyu@gmail.com>
Date: Tue, 30 Mar 2021 09:58:12 +0900
Subject: [PATCH 57/75] ODROID-C4: arm64/dts: Move gpio-line-names into the
 periphs_pinctrl node

In general, the gpio-line-names property should belong to the node
whereby the gpio-controller property by the related document written.
But from now, at least for the Odroid boards based on AMLogic SoC,
the property has to be in the periphs_pinctrl node so that the
function for assigning the line names recognizes that.

Signed-off-by: Deokgyu Yang <secugyu@gmail.com>
Change-Id: I328b47f88c0466de8657997d2c097c7f6fae33d7
---
 .../boot/dts/amlogic/meson-sm1-odroid-c4.dts  | 44 +++++++++++++++++++
 1 file changed, 44 insertions(+)

diff --git a/arch/arm64/boot/dts/amlogic/meson-sm1-odroid-c4.dts b/arch/arm64/boot/dts/amlogic/meson-sm1-odroid-c4.dts
index ca7baefa7a15..cb4c818e9e96 100644
--- a/arch/arm64/boot/dts/amlogic/meson-sm1-odroid-c4.dts
+++ b/arch/arm64/boot/dts/amlogic/meson-sm1-odroid-c4.dts
@@ -70,6 +70,50 @@ opp-1908000000 {
 	};
 };
 
+&periphs_pinctrl {
+	gpio-line-names =
+		/* GPIOZ */
+		"", "", "", "", "", "", "", "",
+		"", "", "", "", "", "", "", "",
+		/* GPIOH */
+		"", "", "", "", "",
+		"PIN_36", /* GPIOH_5 */
+		"PIN_26", /* GPIOH_6 */
+		"PIN_32", /* GPIOH_7 */
+		"",
+		/* BOOT */
+		"", "", "", "", "", "", "", "",
+		"", "", "", "", "", "", "", "",
+		/* GPIOC */
+		"", "", "", "", "", "", "", "",
+		/* GPIOA */
+		"", "", "", "", "", "", "", "",
+		"", "", "", "", "", "",
+		"PIN_27", /* GPIOA_14 */
+		"PIN_28", /* GPIOA_15 */
+		/* GPIOX */
+		"PIN_16", /* GPIOX_0 */
+		"PIN_18", /* GPIOX_1 */
+		"PIN_22", /* GPIOX_2 */
+		"PIN_11", /* GPIOX_3 */
+		"PIN_13", /* GPIOX_4 */
+		"PIN_7",  /* GPIOX_5 */
+		"PIN_33", /* GPIOX_6 */
+		"PIN_15", /* GPIOX_7 */
+		"PIN_19", /* GPIOX_8 */
+		"PIN_21", /* GPIOX_9 */
+		"PIN_24", /* GPIOX_10 */
+		"PIN_23", /* GPIOX_11 */
+		"PIN_8",  /* GPIOX_12 */
+		"PIN_10", /* GPIOX_13 */
+		"PIN_29", /* GPIOX_14 */
+		"PIN_31", /* GPIOX_15 */
+		"PIN_12", /* GPIOX_16 */
+		"PIN_3",  /* GPIOX_17 */
+		"PIN_5",  /* GPIOX_18 */
+		"PIN_35"; /* GPIOX_19 */
+};
+
 &gpio {
 	/*
 	 * WARNING: The USB Hub on the Odroid-C4 needs a reset signal
-- 
2.25.1

